src/tiled/abstractobjecttool.cpp

```diff
@@ -276,6 +288,24 @@ void AbstractObjectTool::showContextMenu(MapObjectItem *clickedObjectItem,
     Utils::setThemeIcon(removeAction, "edit-delete");
     Utils::setThemeIcon(propertiesAction, "document-properties");
 
+    QList<MapObjectItem*> underlyingObjects = listOfObjectItemsAt(scenePos);
+    if (underlyingObjects.size() > 1) {
+        menu.addSeparator();
+
+        QMenu *selectUnderlyingMenu = menu.addMenu(tr("Select an underlying object"));
```

Comment:
My first thought is that indeed it would be more logical to only supports this alternative menu in the ObjectSelectionTool.